pkg/apis/extensions/deep_copy_generated.go

```diff
@@ -24,836 +24,1054 @@ import (
-		for i := range in {
-			if err := DeepCopy_extensions_CustomMetricTarget(in[i], &(*out)[i], c); err != nil {
-				return err
+func DeepCopy_extensions_CustomMetricTargetList(in interface{}, out interface{}, c *conversion.Cloner) error {
+	{
+		in := in.(*CustomMetricTargetList)
+		out := out.(*CustomMetricTargetList)
+		if in.Items != nil {
+			in, out := in.Items, &out.Items
+			*out = make([]CustomMetricTarget, len(in))
+			for i := range in {
+				if err := DeepCopy_extensions_CustomMetricTarget(&in[i], &(*out)[i], c); err != nil {
+					return err
+				}
 			}
+		} else {
+			out.Items = nil
 		}
-	} else {
-		out.Items = nil
+		return nil
 	}
-	return nil
 }
 
-func DeepCopy_extensions_DaemonSet(in DaemonSet, out *DaemonSet, c *conversion.Cloner) error {
-	out.TypeMeta = in.TypeMeta
-	if err := api.DeepCopy_api_ObjectMeta(in.ObjectMeta, &out.ObjectMeta, c); err != nil {
-		return err
-	}
-	if err := DeepCopy_extensions_DaemonSetSpec(in.Spec, &out.Spec, c); err != nil {
-		return err
+func DeepCopy_extensions_DaemonSet(in interface{}, out interface{}, c *conversion.Cloner) error {
+	{
+		in := in.(*DaemonSet)
+		out := out.(*DaemonSet)
+		out.TypeMeta = in.TypeMeta
+		if err := api.DeepCopy_api_ObjectMeta(&in.ObjectMeta, &out.ObjectMeta, c); err != nil {
+			return err
+		}
+		if err := DeepCopy_extensions_DaemonSetSpec(&in.Spec, &out.Spec, c); err != nil {
+			return err
+		}
+		out.Status = in.Status
+		return nil
 	}
-	out.Status = in.Status
-	return nil
 }
 
-func DeepCopy_extensions_DaemonSetList(in DaemonSetList, out *DaemonSetList, c *conversion.Cloner) error {
-	out.TypeMeta = in.TypeMeta
-	out.ListMeta = in.ListMeta
-	if in.Items != nil {
-		in, out := in.Items, &out.Items
-		*out = make([]DaemonSet, len(in))
-		for i := range in {
-			if err := DeepCopy_extensions_DaemonSet(in[i], &(*out)[i], c); err != nil {
-				return err
+func DeepCopy_extensions_DaemonSetList(in interface{}, out interface{}, c *conversion.Cloner) error {
+	{
+		in := in.(*DaemonSetList)
+		out := out.(*DaemonSetList)
+		out.TypeMeta = in.TypeMeta
+		out.ListMeta = in.ListMeta
+		if in.Items != nil {
+			in, out := in.Items, &out.Items
+			*out = make([]DaemonSet, len(in))
+			for i := range in {
+				if err := DeepCopy_extensions_DaemonSet(&in[i], &(*out)[i], c); err != nil {
+					return err
+				}
 			}
+		} else {
+			out.Items = nil
 		}
-	} else {
-		out.Items = nil
+		return nil
 	}
-	return nil
 }
 
-func DeepCopy_extensions_DaemonSetSpec(in DaemonSetSpec, out *DaemonSetSpec, c *conversion.Cloner) error {
-	if in.Selector != nil {
-		in, out := in.Selector, &out.Selector
-		*out = new(unversioned.LabelSelector)
-		if err := unversioned.DeepCopy_unversioned_LabelSelector(*in, *out, c); err != nil {
+func DeepCopy_extensions_DaemonSetSpec(in interface{}, out interface{}, c *conversion.Cloner) error {
+	{
+		in := in.(*DaemonSetSpec)
+		out := out.(*DaemonSetSpec)
+		if in.Selector != nil {
+			in, out := in.Selector, &out.Selector
```

Comment:
Looking at the generated code, this construct is somewhat confusing.  This PR changes to pass all args as pointers, which makes the shadowed in, out technique treat both as pointers, except here.  Should this change to pointers for `in` for consistency?